src/rdkafka_topic.c

```diff
@@ -476,63 +476,81 @@ const char *rd_kafka_topic_name (const rd_kafka_topic_t *app_rkt) {
 }
 
 
-
-
-
 /**
- * @brief Update the leader for a topic+partition.
- * @returns 1 if the leader was changed, else 0, or -1 if leader is unknown.
- *
- * @locks rd_kafka_topic_wrlock(rkt) and rd_kafka_toppar_lock(rktp)
+ * @brief Update the broker for a topic+partition.
+ * 
+ * @param broker_id The id of the broker to associate the toppar with.
+ * @param rkb A reference to the broker to delegate to (must match
+ *        broker_id) or NULL if the toppar should be undelegated for
+ *        any reason.
+ * @param is_leader rd_true if broker_id is the leader, else rd_false.
+ * 
+ * @returns 1 if the broker delegation was changed, -1 if the broker
+ *          delegation was changed and is now undelegated, else 0.
+ * 
+ * @locks caller must have rd_kafka_toppar_lock(rktp)
  * @locality any
  */
-int rd_kafka_toppar_leader_update (rd_kafka_toppar_t *rktp,
-                                   int32_t leader_id, rd_kafka_broker_t *rkb) {
-
-        rktp->rktp_leader_id = leader_id;
-        if (rktp->rktp_leader_id != leader_id) {
-                rd_kafka_dbg(rktp->rktp_rkt->rkt_rk, TOPIC, "TOPICUPD",
-                             "Topic %s [%"PRId32"] migrated from "
-                             "leader %"PRId32" to %"PRId32,
-                             rktp->rktp_rkt->rkt_topic->str,
-                             rktp->rktp_partition,
-                             rktp->rktp_leader_id, leader_id);
-                rktp->rktp_leader_id = leader_id;
-        }
+int rd_kafka_toppar_broker_update (rd_kafka_toppar_t *rktp,
+                                   int32_t broker_id,
+                                   rd_kafka_broker_t *rkb,
+                                   rd_bool_t is_leader) {
 
-	if (!rkb) {
-		int had_leader = rktp->rktp_leader ? 1 : 0;
-
-		rd_kafka_toppar_broker_delegate(rktp, NULL, 0);
+        rktp->rktp_broker_id = broker_id;
+        if (is_leader)
+                rktp->rktp_leader_id = broker_id;
 
-		return had_leader ? -1 : 0;
+	if (!rkb) {
+		int had_broker = rktp->rktp_broker ? 1 : 0;
+		rd_kafka_toppar_broker_delegate(rktp, NULL);
+		return had_broker ? -1 : 0;
 	}
 
-
-	if (rktp->rktp_leader) {
-		if (rktp->rktp_leader == rkb) {
+	if (rktp->rktp_broker) {
+		if (rktp->rktp_broker == rkb) {
 			/* No change in broker */
 			return 0;
 		}
 
 		rd_kafka_dbg(rktp->rktp_rkt->rkt_rk, TOPIC, "TOPICUPD",
-			     "Topic %s [%"PRId32"] migrated from "
-			     "broker %"PRId32" to %"PRId32,
+			     "Topic %s [%"PRId32"]: migrating from "
```

Comment:
When changing a log/dbg thing that has tabs for indent, reindent the whole thing with spaces so at least this one call is consistently indented.